src/Agent.Worker/Build/BuildServer.cs

```diff
@@ -9,28 +9,63 @@
 
 namespace Microsoft.VisualStudio.Services.Agent.Worker.Build
 {
-    public class BuildServer
+    [ServiceLocator(Default = typeof(BuildServer))]
+    public interface IBuildServer : IAgentService
     {
-        private readonly Build2.BuildHttpClient _buildHttpClient;
-        private Guid _projectId;
+        Task ConnectAsync(VssConnection connection);
+        Task<Build2.BuildArtifact> AssociateArtifact(Guid projectId, int buildId, string name, string type, string data, Dictionary<string, string> propertiesDictionary, CancellationToken cancellationToken);
+        Task<Build2.Build> UpdateBuildNumber(Guid projectId, int buildId, string buildNumber, CancellationToken cancellationToken);
+        Task<IEnumerable<string>> AddBuildTag(Guid projectId, int buildId, string buildTag, CancellationToken cancellationToken);
+    }
+
+    public sealed class BuildServer : AgentService, IBuildServer
+    {
+        private bool _hasConnection;
+        private VssConnection _connection;
+        private Build2.BuildHttpClient _buildHttpClient;
 
-        public BuildServer(VssConnection connection, Guid projectId)
+        public async Task ConnectAsync(VssConnection connection)
```

Comment:
Why roll our own retry? Ted claimed the http client retries if settings are passed correctly. Are we passing the settings wrong?

Reply:
Agreed.  What are we doing with this PR?